src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource.cs

```diff
@@ -206,7 +206,7 @@ public void Message(string Message)
         /// Events from DiagnosticSource can be forwarded to EventSource using this event.
         /// </summary>
         [Event(2, Keywords = Keywords.Events)]
-        private void Event(string SourceName, string EventName, IEnumerable<KeyValuePair<string, string>> Arguments)
+        private void Event(string? SourceName, string EventName, IEnumerable<KeyValuePair<string, string>> Arguments)
```

Comment:
The event name in all on all of these Event's can clearly take null in all the arguments, it doesn't matter much as it is an internal class, but for correctness it would make sense. All these end up calling https://source.dot.net/#System.Private.CoreLib/shared/System/Diagnostics/Tracing/EventSource.cs,1876 with all of the args as an `object?[]` array

Reply:
yes, but I was not tried to evaluate nullability for private methods unless compiler complaining. Let me know if that is Ok.

I have another question for this case,  object?[] array means array cannot be null, but if we make all variable arguments the array will be null right? what happens in that case

Reply:
Correct, the array can have null values, but the array itself, can't be null.